monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -1352,10 +1352,21 @@ abstract class Observable[+A] extends Serializable { self =>
     * producing and concatenating observables along the way.
     *
     * It's the combination between [[scan]] and [[flatMap]].
+    *
+    * @see [[flatScan0]] for the version that emits seed element at the beginning
     */
   final def flatScan[R](seed: => R)(op: (R, A) => Observable[R]): Observable[R] =
     new FlatScanObservable[A, R](self, seed _, op, delayErrors = false)
 
+  /** Applies a binary operator to a start value and to elements
+    * produced by the source observable, going from left to right,
+    * producing and concatenating observables along the way.
+    *
+    * It's the combination between [[scan0]] and [[flatMap]].
+    */
+  final def flatScan0[R](seed: => R)(op: (R, A) => Observable[R]): Observable[R] =
+    Observable.eval(seed).flatMap(s => s +: flatScan(s)(op))
```

Comment:
This is fine, but I would use `++` because it is more efficient and stack/memory safe for that matter.

Reply:
Since `seed` is by-name parameter, I am assuming it can be effectful, in which case it should be evaluated once, as in regular `scan`. I don't think it can be done without `flatMap`.

Reply:
Sorry, I just skimmed the code without thinking. You actually need `flatMap` here. So it's fine.